internal/pkg/gateway/commit/chaincodeeventnotifier.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/core/ledger"
+)
+
+type BlockChaincodeEvents struct {
+	BlockNumber uint64
+	Events      []*peer.ChaincodeEvent
+}
+
+type chaincodeEventListenerSet map[*chaincodeEventListener]bool
```

Comment:
nit: standard usage for `set`  (in other file as well)
```suggestion
type chaincodeEventListenerSet map[*chaincodeEventListener]struct{}
```